edtLHD/src/main/java/fr/univtln/lhd/facade/Schedule.java

```diff
@@ -0,0 +1,149 @@
+            try {
+                List<Slot> allSlotOfGroup = dao.getSlotOfGroup(group);
+                slotList.addAll(allSlotOfGroup);
+            } catch (SQLException e) {
+                log.error(e.getMessage());
+            }
+        }
+        slotList = schedule.filterEverySlotOutsideTimerange(slotList, timerange);
+        return slotList;
+    }
+
+
+    /**
+     * Take a Group and two LocalDate(start/end) and return every slot that the group is
+     * in, in the timerange of the two LocalDate
+     * The group must be persisted or will return an empty array
+     *
+     * @param group     a persisted group
+     * @param timeStart the start of the timerange
+     * @param timeEnd   the end of the timerange
+     * @return a List of slot that is in the timerange with this group
+     */
+    public static List<Slot> getSchedule(Group group, LocalDate timeStart, LocalDate timeEnd) {
+        Interval timerange = schedule.getIntervalOf(timeStart, timeEnd);
+        return Schedule.getSchedule(group, timerange);
+    }
+
+    /**
+     * Take a Group and an Interval and return a List of the slot with this group
+     * inside the interval
+     * The group must be persisted or will return an empty array
+     *
+     * @param group     a persisted group
+     * @param timerange an Interval
+     * @return a List of slot that is in the timerange and with this group
+     */
+    public static List<Slot> getSchedule(Group group, Interval timerange) {
+        if (group.getId() < 0) {
+            return new ArrayList<>();
+        }//Might be an edge case
+        SlotDAO dao = SlotDAO.getInstance();
+        List<Slot> allSlotOfGroup = new ArrayList<>();
+        try {
+            allSlotOfGroup = dao.getSlotOfGroup(group);
+        } catch (SQLException e) {
+            log.error(e.getMessage());
+        }
+        return schedule.filterEverySlotOutsideTimerange(allSlotOfGroup, timerange);
+    }
+
+
+    /**
+     * Take a list of Slot and return another List of slot inside the timerange
+     *
+     * @param slotList  list of all the slot
+     * @param timerange an Intervalle of the slot to keep
+     * @return all the slot of slotList that was inside the timerange
+     */
+    private List<Slot> filterEverySlotOutsideTimerange(List<Slot> slotList, Interval timerange) {
+        List<Slot> slotIntimerange = new ArrayList<>();
+        if (slotList.isEmpty()) {
+            return slotIntimerange;
+        }
+        for (Slot slot :
+                slotList) {
+            if (timerange.encloses(slot.getTimeRange())) {
+                slotIntimerange.add(slot);
+            }
+        }
+        return slotIntimerange;
+    }
+
+    /**
+     * Take two LocalDate and return an Interval between the two time
+     * to be noted LocalDate doesn't have a time therefor each instant
+     * start at the beginning of the day 00h00
+     * If the two time are on the same day, the all day is return
```

Comment:
Typo in comments

Reply:
where, timerange is not a real word but It seems understandable to me

Reply:
whole vs all